python/mspasspy/db/database.py

```diff
@@ -97,7 +122,7 @@ def read_data(self, object_id, collection, load_history=True, metadata_schema=No
         md['source_time'] = source['time']
         md['source_magnitude'] = source['magnitude']
 
-        if object['dtype'] == 'TimeSeries':
+        if collection == 'wf_TimeSeries':
```

Comment:
The collection used here is just to identify which data type the objectId corresponds to. We can change the argument to 'data_type' if you want.

Reply:
What I really mean here is that when users call this `read_data` method, by default we assume he is trying to read data from one of the `wf` collections. If a `collection` argument is not given, we should just call `DatabaseSchema.default_name('wf')` to determine whether "wf_TimeSeries" or "wf_Seismogram" is the default defined by the yaml file.

Reply:
btw, I mentioned this idea of adding a type method below https://github.com/wangyinz/mspass/pull/98#discussion_r549477810

Reply:
I'm still not sure about this change. From my point of view, setting an explicit object_type when using the method is more clear and easier to use than setting the default type in the schema.

Reply:
Well, data_type is indeed part of the schema. For example, we definitely don't want a user to put `Seismogram` into `wf_TimeSeries`, otherwise there is no point defining `wf_TimeSeries` and `wf_Seismogram` in the first place.

Reply:
right now, we only specify object_type in the parameters, a 'TimeSeries' object_type will be implicitly bound to 'wf_TimeSeries'. And the default is 'TimeSeries'. I think it's already easy to understand and to be used.

Reply:
Note that we have more `wf_` collections than just `wf_TimeSeries` and `wf_Seismogram`. For example, in Gary's raw data handling module that he's working on right now, he is going to define a `wf_miniseed` collection. In that case, this method won't be support operations on that `wf_miniseed` collection albeit most of the attributes there are the same with a `wf_TimeSeries`. I think user-defined `wf_` collection like this will be very common, and that's the reason why we have this `DatabaseSchema.default_name('wf')` method. Basically, this method will return `wf_miniseed` in that example.

Reply:
I see. But I think right now our solution already supports this. The wf collection is defined as
`wf_collection = read_metadata_def.collection('_id')`, in this way, it's the user's responsibility to assgin the right collection name in the YAML file.

Reply:
Actually, that reveals another hole in our schema design. What if `DatabaseSchema.default_name('wf')` and `read_metadata_def.collection('_id')` points to different collections. We have the `set_default` and `unset_default` method that can easily change the former at runtime, but we don't have equivalents in the latter. Maybe we should add that? Supposedly, user will want to interact with more than one collection in a job (e.g., wf_miniseed and wf_TimeSeries).

Reply:
In this way, users just need to call set_metadata_schema and set_database_schema when using different schemas.

Reply:
Yeah, that works, but I think we do need a method to modify the schema at runtime, otherwise the set_metadata_schema and set_database_schema will depend on a number of different yaml files, which seems not very convenient. I guess for now, this is fine.

Reply:
done